### PR TITLE
Make Keycloak fail with an error when the persisted build options differs from those provide

### LINKED ISSUE #32717: Make it explicit which options are needed when using optimized image with the Operator
### Description

Currently, the guide does not explicitly warn the user that when using custom optimized image, some build time options are actually required to be baked in the image.

This probably concerns only health and metrics.

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by Pepo48**
* PropertyException is now thrown instead of a warning
* Operator guides clarification around health and metrics options

Closes: #32717

### CODE REVIEW

**Comment by shawkins** [comment]
@Pepo48 @vmuzikar I think we also discussed having the server throw an exception instead of emitting a warning in the case where build time options seem to change. Was something like that going to be included here?

**Comment by Pepo48** [comment]
@shawkins you're correct and I already have it prepared under different branch (I might've misunderstood it should be part of this PR). Let me do a cherrypick.

**Review by shawkins — COMMENTED** [review]
Generally looks good, just some thoughts about the exception.

File: docs/guides/operator/customizing-keycloak.adoc [hunk 1]
```diff
@@ -21,6 +21,7 @@ When using the default {project_name} image, the server will perform a costly re
 To avoid this delay, you can provide a custom image with the augmentation built-in from the build time of the image.
 
 With a custom image, you can also specify the Keycloak _build-time_ configurations and extensions during the build of the container.
+Options like `health-enabled` and `metrics-enabled` are good examples of why this is sometimes necessary. By default, these options are disabled. To make these endpoints available, you can either use the `ENV` instruction in the Containerfile or specify them as parameters of the {project_name}'s `build` command.
```
Comment (vmuzikar) [inline docs/guides/operator/customizing-keycloak.adoc:24]: I'd be even more explicit here. I'd state that when using optimized custom image, `health-enabled` and `metrics-enabled` need to be explicitly set in the Containerfile. And maybe make it a warning even.

**Comment by vmuzikar** [comment]
We also need some mention in the upgrading guide.

File: docs/documentation/upgrading/topics/changes/changes-26_0_0.adoc [hunk 2]
```diff
@@ -325,3 +325,7 @@ Keycloak JS now utilizes the Web Crypto API to calculate the SHA-256 digests nee
 - `createRegisterUrl()`
 
 Make sure to update your code to `await` these methods.
+
+= Stricter startup behavior for CLI build-time options
```
Comment (vmuzikar) [inline docs/documentation/upgrading/topics/changes/changes-26_0_0.adoc:329]:
```suggestion
= Stricter startup behavior for build-time options
```

**Review by vmuzikar — CHANGES_REQUESTED** [review]

**Review by vmuzikar — APPROVED** [review]
@Pepo48 Thank you for the changes, LGTM!

**Review by shawkins — APPROVED** [review]
LGTM, thanks @Pepo48

**Pull request merged**